galeracluster/source/selinux.rst

```diff
@@ -0,0 +1,128 @@
+
+To disable SELinux, complete the following steps:
+
+#. Using ``semanage``, open the relevant ports:
+
+   .. code-block:: console
+
+      # semanage port -a -t mysqld_port_t -p tcp 4567
+      # semanage port -a -t mysqld_port_t -p tcp 4568
+      # semanage port -a -t mysqld_port_t -p tcp 4444
+
+   SELinux already opens the standard MySQL port ``3306``.  In the event that you use UDP in your cluster, you also need to open ``4567`` to those connections.
+
+   .. code-block:: console
+
+      # semanage port -a -t mysqld_port_t -p udp 4567
+
+#. Set SELinux to permissive mode for the database server.
+
+   .. code-block:: console
+
+      # semanage permissive -a mysqld_t
+
+SELinux now permits the database server to function on the server and no longer blocks the node from network connectivity with the cluster.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Defining the SELinux Policy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`define-selinux-policy`:
+
+While SELinux remains in permissive mode, it continues to log activity from the database server.  In order for it to understand normal operation for the database, you need to start the database and generate routine events for SELinux to see.
+
+For servers that use ``init``, start the database with the following command:
+
+.. code-block:: console
+		
+   # service mysql start
+
+For servers that use ``systemd``, instead run this command:
+
+.. code-block:: console
+
+   # systemctl mysql start
+
+You can now begin to create events for SELinux to log.  There are many ways to go about this, including:
+
+- Deleting the write-set cache forces the node to restore itself using a :term:`State Snapshot Transfer`.  The write-set cache is typically in the datadir and is named ``galera.cache`.
+
+- Stop the node, then make changes on another node before starting it again.  Not being that far behind, the node updates itself using an :term:`Incremental State Transfer`.
+
+- Using :ref:`wsrep_desync <wsrep_desync>`, you can change the node state, triggering the notification command defined by :ref:`wsrep_notify_cmd <wsrep_notify_cmd>`. 
+
+- Using :term:`Galera Arbitrator` to run a backup.
+
+When you feel you have generated sufficient events for the log, you can begin work creating the policy and turning SELinux back on.
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Enabling an SELinux Policy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`enable-selinux`:
+
+Generating an SELinux policy requires that you search log events for the relevant information and pipe it to the ``audit2allow`` utility, creating a ``galera.te`` file to load into SELinux.
+
+To generate and load an SELinux policy for Galera Cluster, complete the following steps:
+
+#. Using ``fgrep`` and ``audit2allow``, create a textease file with the policy information.
```

Comment:
Came up in the research.  Textease is what they call the files with the .te extension.  It's some kind of obscure database file format.